lib/gc/Transforms/TilingUsingInterfaceX.cpp

```diff
@@ -296,6 +309,57 @@ mlir::scfX::tileAndFuseProducerOfSlice(RewriterBase &rewriter,
         tileAndFuseProducerOfSliceImpl(rewriter, sliceOp, outerLoops);
     if (!fuseProducerResult)
       return std::nullopt;
+
+    // Cache old listener.
+    OpBuilder::Listener *oldListener = rewriter.getListener();
+    // Set new listener.
+    ErasedOpListener *newListener = new ErasedOpListener();
```

Comment:
Shall we call the oldListener? If `oldListener` does something important in the callback, `rewriter.setListener(newListener);` will override it. Maybe we should hold a `Listener*` in your `ErasedOpListener ` to pass down the callback to `oldListener `

Reply:
Yes, exactly. Add old listener hook now.